lib/src/fnc/parse.rs

```diff
@@ -3,60 +3,47 @@ pub mod email {
 	use crate::ctx::Context;
 	use crate::err::Error;
 	use crate::sql::value::Value;
-	use once_cell::sync::Lazy;
-	use regex::Regex;
-
-	#[rustfmt::skip] static USER_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r"^(?i)[a-z0-9.!#$%&'*+/=?^_`{|}~-]+\z").unwrap());
-	#[rustfmt::skip] static HOST_RE: Lazy<Regex> = Lazy::new(|| Regex::new(r"(?i)^[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?(?:\.[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?)*$",).unwrap());
+	use addr::email::Host;
 
 	pub fn domain(_: &Context, mut args: Vec<Value>) -> Result<Value, Error> {
 		// Convert to a String
 		let val = args.remove(0).as_string();
-		// Check if value is empty
-		if val.is_empty() {
-			return Ok(Value::None);
-		}
-		// Ensure the value contains @
-		if !val.contains('@') {
-			return Ok(Value::None);
-		}
-		// Reverse split the value by @
-		let parts: Vec<&str> = val.rsplitn(2, '@').collect();
-		// Check the first part matches
-		if !USER_RE.is_match(parts[1]) {
-			return Ok(Value::None);
-		}
-		// Check the second part matches
-		if !HOST_RE.is_match(parts[0]) {
-			return Ok(Value::None);
+		// Parse the email address
+		match addr::parse_email_address(&val) {
+			// Return the host part
+			Ok(v) => match v.host() {
+				Host::Domain(name) => Ok(name.as_str().into()),
+				Host::IpAddr(ip_addr) => Ok(ip_addr.to_string().into()),
```

Comment:
@tobiemh Considering that the function is called `parse::email::domain` returning an IP address here doesn't feel right. I think either renaming it to something like `parse::email::host` or returning `Value::None` might be better. I left it like this for now because this is what is equivalent to the previous code. Email addresses containing IP addresses were not accepted though. How do you want me to handle this?

Reply:
@rushmorem, good idea - let's change this function to `parse::email::host()`. That does make sense yes! Happy to do it in a different PR if that is easier?